design/baremetal-operator/user-defined-parameters.md

```diff
@@ -0,0 +1,170 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# Add Support for User-defined Parameters
+
+## Status
+
+provisional
+
+## Summary
+
+Add an optional field `extraParametersName` to accept a `Secret` object
+where to allow user to set vendor specific configuration.
+
+## Motivation
+
+When deploying BM nodes, some vendor's BMC will require additional parameters
+to adapt to certain situations. For example:
+
+1. When FIPS mode is enabled on the host running the ironic service, the
+   version of SNMP protocol of iRMC driver must be v3. Therefore, additional
+   parameters are needed to specify the version and to provide the corresponding
+   credential information.
```

Comment:
I could potentially be convinced that the passwords for SNMP should be read as separate fields from the existing BMC credentials Secret, if I were convinced that drivers needing to access SNMP (and the passwords being different from the BMC password) were a thing that made sense generally.
The v3 part can easily be inferred by the iRMC BMC plugin based on the presence of the keys.

Reply:
If it's just a protocol version though, it can be a parameter in the BMC address: `irmc://ip.address?snmp_version=3`